### PR TITLE
Added Code Sample and manual for S3 Object DataNode

### LINKED ISSUE #775: Document S3DataNode
**Description**
Add documentation for S3 data node.

### PR DESCRIPTION
**Pull request by Forchapeatl**
Part of https://github.com/Avaiga/taipy/issues/536

### CODE REVIEW

File: docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#Upload and retrive raw data from existing s3 bucket
```
Comment (jrobinAV) [inline docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py]:
```suggestion
# Upload and retrieve raw data from existing s3 bucket
```

File: docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+input_text = None
+s3_result = None
+
+
+page = """
+message: <|{input_text}|input|>
+<|submit|button|on_action=upload_scenario|>
+
+Retrived : <|{s3_result}|text|>
```
Comment (jrobinAV) [inline docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py]:
```suggestion
Retrieved : <|{s3_result}|text|>
```

File: docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+input_text_data_node_cfg = Config.configure_data_node(id="input_text")
+s3_result_data_node_cfg = Config.configure_data_node(id="s3_result")
+store_data_task_cfg = Config.configure_task("upload", get_s3_data, input=[s3_object_cfg,input_text_data_node_cfg], output=[s3_result_data_node_cfg])
+scenario_cfg = Config.configure_scenario("scenario", task_configs=[store_data_task_cfg])
+
+if __name__ == "__main__":
+
+    Core().run()
+    scenario = tp.create_scenario(scenario_cfg)
+    Gui(page).run(debug=True)
```
Comment (trgiangdo) [inline docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py]: The example should be minimal, which means only containing the configuration.

You can add an example on how to read/write S3ObjectDataNode on the `docs/manuals/core/entities/data-node-mgt.md` page.

File: docs/manuals/core/config/data-node-config.md [hunk 4]
```diff
@@ -76,7 +77,7 @@ Taipy proposes predefined *data nodes* corresponding to the most popular
 does not need to spend much time configuring the *storage types* or the
 *query system*. A predefined data node will often satisfy the user's
 required format: pickle, CSV, SQL table, MongoDB collection, Excel sheet,
-etc.
+Amazon Web Service S3 Object etc.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
Amazon Web Service S3 Object, etc.
```

File: docs/manuals/core/config/data-node-config.md [hunk 5]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+S3 bucket. To add a new *S3Object* data node configuration, the
+`Config.configure_s3_object_data_node()^` method can be used. In addition to
+the generic parameters described in the [Data node configuration](data-node-config.md)
+section, multiple parameters can be provided.
+
+- _**aws_access_key**_ represents the Amazon Web Services identity account.
+- _**aws_secret_access_key**_ represents the Amazon Web Services access key to authenticate programmatic requests.
+- _**aws_region**_ represnets the geographic area where Amazon Web Services (AWS) infrastructure is located.
+- _**aws_s3_bucket_name**_ represnts the name aof the Amazon Web Service S3 bucket.
+- _**aws_s3_object_key**_ represents the name of the object(file) that has to be read or written.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws_s3_object_key**_ represents the name of the object (file) that needs to be read or written.
```

File: docs/manuals/core/config/data-node-config.md [hunk 6]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+- _**aws _s3_object_parameters**_  represents additional arguments to be passed to interact with the AWS service.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws _s3_object_parameters**_ represents additional arguments to be passed to interact with the AWS service.
```

File: docs/manuals/core/config/data-node-config.md [hunk 7]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+
+In this example, we configure an *s3_object* data node with the id "my_s3_objec":
+
+- Its scope is the default value `SCENARIO`.
+- The object_key name  is "taipy_object"
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- The object_key name is "taipy_object"
```

File: docs/manuals/core/config/data-node-config.md [hunk 8]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+section, multiple parameters can be provided.
+
+- _**aws_access_key**_ represents the Amazon Web Services identity account.
+- _**aws_secret_access_key**_ represents the Amazon Web Services access key to authenticate programmatic requests.
+- _**aws_region**_ represnets the geographic area where Amazon Web Services (AWS) infrastructure is located.
+- _**aws_s3_bucket_name**_ represnts the name aof the Amazon Web Service S3 bucket.
+- _**aws_s3_object_key**_ represents the name of the object(file) that has to be read or written.
+- _**aws _s3_object_parameters**_  represents additional arguments to be passed to interact with the AWS service.
+
+In this example, we configure an *s3_object* data node with the id "my_s3_objec":
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
In this example, we configure a *S3Object* data node with the id "my_s3_object":
```
For consistency

File: docs/manuals/core/config/data-node-config.md [hunk 9]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
     with parameter `storage_type="in_memory"`.
 
+## Amazon Web Service S3 Object
+A `S3ObjectDataNode^` is a specific data node used to model data stored in a
+S3 bucket. To add a new *S3Object* data node configuration, the
+`Config.configure_s3_object_data_node()^` method can be used. In addition to
+the generic parameters described in the [Data node configuration](data-node-config.md)
+section, multiple parameters can be provided.
+
+- _**aws_access_key**_ represents the Amazon Web Services identity account.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws_access_key**_ represents the Amazon Web Services (AWS) identity account.
```

The abbreviation should be defined at the first appearance.

File: docs/manuals/core/config/data-node-config.md [hunk 10]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+- _**aws_secret_access_key**_ represents the Amazon Web Services access key to authenticate programmatic requests.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws_secret_access_key**_ represents the AWS access key to authenticate programmatic requests.
```

File: docs/manuals/core/config/data-node-config.md [hunk 11]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+- _**aws_region**_ represnets the geographic area where Amazon Web Services (AWS) infrastructure is located.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws_region**_ represnets the geographic area where the AWS infrastructure is located.
```

File: docs/manuals/core/config/data-node-config.md [hunk 12]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+- _**aws_s3_bucket_name**_ represnts the name aof the Amazon Web Service S3 bucket.
```
Comment (trgiangdo) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- _**aws_s3_bucket_name**_ represnts the name of the AWS S3 bucket.
```

**Review by trgiangdo — COMMENTED** [review]
Thank you @Forchapeatl. I have some small comments, and concerning the configuration example.

File: docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py (hunk 3 above)
Comment (jrobinAV) [reply docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py]: I agree. 
- I believe there is no need to run any Gui here.
- I believe we don't need to configure a scenario here. 
    - We can just configure a Global s3 data node using `Config.configure_s3_data_node(...)`.
    - We can instantiate the data node using `taipy.create_global_data_node(...)`
    - We can explain how to write data and how to read data using 
    `datanode.write(...)` and `datanode.read()`

**Comment by trgiangdo** [comment]
@Forchapeatl Thank you once again for your contribution. We highly appreciate your work and eagerly collaborating with you on this PR.

If you could provide an update or let us know if you need more time to answer the unresolved Comments, that would be greatly appreciated.

**Comment by Forchapeatl** [comment]
> @Forchapeatl Thank you once again for your contribution. We highly appreciate your work and eagerly collaborating with you on this PR.
> 
> If you could provide an update or let us know if you need more time to answer the unresolved Comments, that would be greatly appreciated.

I just need more time . Please I will work on this before this week ends.

**Comment by Forchapeatl** [comment]
Hello  @trgiangdo and @jrobinAV , thank you for the reviews . The requested changes have been made.

File: docs/manuals/core/config/data-node-config.md [hunk 13]
```diff
@@ -839,6 +840,32 @@ The scope is `SCENARIO` (default value), and default data is provided.
...
+- _**aws_secret_access_key**_ represents the AWS access key to authenticate programmatic requests.
+- _**aws_region**_ represnets the geographic area where the AWS infrastructure is located.
+- _**aws_s3_bucket_name**_ represnts the name of the AWS S3 bucket.
+- _**aws_s3_object_key**_ represents the name of the object (file) that needs to be read or written.
+- _**aws _s3_object_parameters**_ represents additional arguments to be passed to interact with the AWS service.
+
+In this example, we configure a *s3_object* data node with the id "my_s3_object":
+
+- Its scope is the default value `SCENARIO`.
+- The object_key name is "taipy_object"
```
Comment (jrobinAV) [inline docs/manuals/core/config/data-node-config.md]:
```suggestion
- The object_key name is "taipy_object".
```

**Review by jrobinAV — APPROVED** [review]

File: docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py [hunk 14]
```diff
@@ -0,0 +1,23 @@
+# Let's welcome users with our s3 object
```
Comment (trgiangdo) [inline docs/manuals/core/config/code_example/data_node_cfg/data-node-config-aws-s3.py]:
```suggestion
```

**Review by trgiangdo — APPROVED** [review]
LGTM. 1 minor comment on the code example, otherwise I believe this PR is ready to be merged

**Pull request merged**